core/src/stores/postgres.rs

```diff
@@ -71,6 +71,92 @@ impl Store for PostgresStore {
         Ok(Project::new_from_id(row_id))
     }
 
+    async fn delete_project(&self, project: &Project) -> Result<()> {
+        let project_id = project.project_id();
+        let pool = self.pool.clone();
+        let mut c = pool.get().await?;
+        let tx = c.transaction().await?;
+
+        // Block executions, Runs & Runs joins
+        let create_proc = r#"
+            CREATE OR REPLACE FUNCTION delete_project_runs(v_project_id BIGINT)
+            RETURNS void AS $$
+            DECLARE
+                run_ids BIGINT[];
+                block_exec_ids BIGINT[];
+            BEGIN
+                -- Store run IDs in an array for the specified project
+                SELECT array_agg(id) INTO run_ids FROM runs WHERE project = v_project_id;
+
+                -- Store block_execution IDs in an array
+                SELECT array_agg(block_execution) INTO block_exec_ids
+                FROM runs_joins 
+                WHERE run = ANY(run_ids);
+
+                -- Delete from runs_joins where run IDs match those in the project
+                DELETE FROM runs_joins WHERE block_execution = ANY(block_exec_ids);
+
+                -- Now delete from block_executions using the stored IDs
+                DELETE FROM block_executions WHERE id = ANY(block_exec_ids);
+
+                -- Finally, delete from runs where run IDs match those in the project
+                DELETE FROM runs WHERE id = ANY(run_ids);
+            END;
+            $$ LANGUAGE plpgsql;
+        "#;
+
+        tx.execute(create_proc, &[]).await?;
+        tx.execute("SELECT delete_project_runs($1)", &[&project_id])
+            .await?;
+
+        // Datasets joins & Datasets points & Datasets
+        let create_proc = r#"
+            CREATE OR REPLACE FUNCTION delete_project_datasets(v_project_id BIGINT)
```

Comment:
Let's move to store.rs as CREATE IF NOT EXISTS :pray:

Reply:
Answered in the other comment, it does not exist 😢